squad/frontend/views.py

```diff
@@ -102,13 +103,26 @@ def build(request, group_slug, project_slug, version):
         'suite',
     ).order_by('-tests_fail', 'suite__slug')
 
-    count_statuses_nofail = len([s for s in statuses if s.has_tests and s.tests_fail == 0])
-    count_statuses_fail = len([s for s in statuses if s.has_tests and s.tests_fail > 0])
+    grouped_statuses = OrderedDict()
+    suite_fails = dict()
+    for s in _statuses:
+        if s.has_tests:
+            if s.tests_fail > 0 and s.suite not in suite_fails.keys():
+                suite_fails[s.suite] = True
+            if s.suite not in grouped_statuses.keys():
+                grouped_statuses[s.suite] = {s.test_run.environment: []}
+            if s.test_run.environment not in grouped_statuses[s.suite].keys():
+                grouped_statuses[s.suite][s.test_run.environment] = []
+            grouped_statuses[s.suite][s.test_run.environment].append(s)
+
+    count_statuses_nofail = len([s for s in _statuses if s.has_tests and s.tests_fail == 0])
```

Comment:
these 2 variables can be also calculated in the loop above to save some cycles.

Reply:
@mwasilew please see the comment in the last remark below as it relates to this one. These counters now need to be removed.